CharlieBackend.AdminPanel/Exceptions/HttpStatusException.cs

```diff
@@ -0,0 +1,26 @@
+﻿using CharlieBackend.Core.DTO.Result;
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+
+namespace CharlieBackend.AdminPanel.Exceptions
+{
+    class HttpStatusException : Exception
+    {
+        private static string GetFormattedMessage(HttpResponseMessage httpResponseMessage)
+        {
+            string stringResponse = httpResponseMessage.Content.ReadAsStringAsync().GetAwaiter().GetResult();
```

Comment:
Try not to use such approach with GetAwaiter().GetResult() because in some circumstances it may result in a deadlock situation. [link](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html)
So please await ReadAsStringAsync() in async context and pass message to constructor of this exception. Try to keep exceptions simple, it shouldn't be aware of httpResponse format. Just pass code and message to it.